### PR TITLE
Node editor

### PR DESCRIPTION
**Pull request by alkaitagi**
Reworked editor UIs.
Hid certain UIs for unauthed.

### CODE REVIEW

**Comment by alkaitagi** [comment]
- Add interactive narrative thread view.

File: lib/widgets/maybe_pop_alert.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import 'package:flutter/material.dart';
+
+class MaybePopAlert extends StatelessWidget {
+  const MaybePopAlert(
+    this.child, {
```
Comment (aabounegm) [inline lib/widgets/maybe_pop_alert.dart:5]: Isn't it idiomatic in Flutter to have `child` as a `required` named parameter?

File: lib/editor/story_editor_screen.dart [hunk 2]
```diff
@@ -131,91 +106,36 @@ class StoryEditorState extends State<StoryEditorScreen> {
...
+            );
+          }),
+          currentIndex: _page,
+          items: {
+            'General': Icons.auto_stories_rounded,
+            'Actors': Icons.person_rounded,
+            'Narrative': Icons.timeline_rounded,
+          }
+              .entries
+              .map((p) => BottomNavigationBarItem(
```
Comment (aabounegm) [inline lib/editor/story_editor_screen.dart:134]: Why do you sometimes use `.map` and sometimes "collection for"? Let's try to make it consistent unless one of them is really more suitable for the specific situation

File: lib/screens/crowdsourcing_screen.dart [hunk 3]
```diff
@@ -21,97 +21,98 @@ class CrowdsourcingScreen extends StatefulWidget {
 }
 
 class _CrowdsourcingScreenState extends State<CrowdsourcingScreen> {
-  late Map<String, TranslationAsset> translations;
-  late final Map<String, TranslationAsset> origins;
-  String language = '';
+  late final translations = <String, TranslationAsset>{};
```
Comment (aabounegm) [inline lib/screens/crowdsourcing_screen.dart:24]: Why is this one `late`?

File: lib/screens/story_screen.dart [hunk 4]
```diff
@@ -21,20 +22,6 @@ class StoryScreen extends StatefulWidget {
 }
 
 class _StoryScreenState extends State<StoryScreen> {
```
Comment (aabounegm) [inline lib/screens/story_screen.dart:24]: I don't see any `setState` in this file. Why is this widget a `StatefulWidget`?

**Review by aabounegm — COMMENTED** [review]
I don't understand why all the Editor-related code is in a folder called `editor`. Are we changing the way the project is structured to be by feature rather than file purpose? By this I mean that I see screens (`Scaffold`s) outside the `screens` folder, and same for other widgets.

There are so many `late final` fields whose only purpose is to link to `widget.something`. It doesn't feel like it saves many keystrokes, and seems to cause problems with type inference. I would rethink that structure, and prefer passing that to the state constructor instead.

I also can't say I understood every single function and widget, but I tested on Android and Chrome and it works. However, I do have the feeling that there is some repetition, and we can come back to that after the upcoming deadlines. Generally speaking, we need to develop a scalable architecture for the code. I don't like how much nullable variables there are.

**Pull request merged**